test/test_bundle_importer.js

```diff
@@ -0,0 +1,189 @@
+          assert.equal(notifications.add.callCount(), 1);
+          assert.equal(notifications.add.lastArguments()[0].level, 'error');
+        });
+
+        it('shows notification before kicking off import', function() {
+          var importStub = utils.makeStubMethod(
+              bundleImporter, 'importBundleDryRun');
+          this._cleanups.push(importStub.reset);
+          bundleImporter._fileReaderOnload(file, {
+            target: { result: '["valid", "json"]' }
+          });
+          assert.equal(notifications.add.callCount(), 1);
+          assert.equal(notifications.add.lastArguments()[0].level, 'important');
+          assert.equal(importStub.callCount(), 1);
+        });
+      });
+    });
+    describe('importBundleDryRun', function() {
+      var unsortedRecords = [
+        { method: 'addRelation' },
+        { method: 'addUnit' },
+        { method: 'addMachines' },
+        { method: 'deploy' },
+        { method: 'addCharm' }
+      ];
+
+      var sortedRecords = [
+        { method: 'addCharm' },
+        { method: 'addMachines' },
+        { method: 'deploy' },
+        { method: 'addUnit' },
+        { method: 'addRelation' }
+      ];
+
+      it('sorts the records then calls to execute', function() {
+        var execute = utils.makeStubMethod(bundleImporter, '_executeDryRun');
+        this._cleanups.push(execute.reset);
+        bundleImporter.importBundleDryRun(unsortedRecords);
+        assert.deepEqual(bundleImporter.recordSet, sortedRecords);
+        assert.equal(execute.callCount(), 1);
+        assert.deepEqual(execute.lastArguments()[0], sortedRecords);
+      });
+    });
+  });
+
+  describe('Changeset generation methods', function() {
+    it('calls to execute supplied records', function() {
+      var called = {
+        addCharm: false,
+        addMachines: false,
+        deploy: false,
+        addUnit: false,
+        addRelation: false
+      };
+      function executor(record, next) {
+        called[record.method] = true;
+        next();
+      }
+      bundleImporter._execute_addCharm = executor;
+      bundleImporter._execute_addMachines = executor;
+      bundleImporter._execute_deploy = executor;
+      bundleImporter._execute_addUnit = executor;
+      bundleImporter._execute_addRelation = executor;
+
+      var sortedRecords = [
+        { method: 'addCharm' },
+        { method: 'addMachines' },
+        { method: 'deploy' },
+        { method: 'addUnit' },
+        { method: 'addRelation' }
+      ];
+      bundleImporter._executeDryRun(sortedRecords);
+      assert.deepEqual(called, {
+        addCharm: true,
+        addMachines: true,
+        deploy: true,
+        addUnit: true,
+        addRelation: true
+      });
+    });
+
+    it('does not fail if unknown record type', function() {
+      var sortedRecords = [
+        { method: 'badMethod' }
+      ];
+      bundleImporter._executeDryRun(sortedRecords);
+    });
+
+    describe('_execute_addCharm', function() {
```

Comment:
These could use an XXX comment with your note about implementing them in the future.